src/compiler/compile/render_dom/wrappers/Element/index.ts

```diff
@@ -1077,7 +1110,7 @@ export default class ElementWrapper extends Wrapper {
 
 			block.chunks.hydrate.push(updater);
 
-			if (has_spread || this.node.is_dynamic_element) {
+			if ((this.node.is_dynamic_element || has_spread) && this.has_dynamic_attribute) {
```

Comment:
I'm having a hard time wrapping my head around whether or not this condition is correct. I think it's incorrect because a dynamic element without spread or dynamic attributes still needs to have the `toggle_class` method applied, because if the element itself changes, the class needs to be reapplied. What does the addition of `this.has_dynamic_attribute` has to do with this condition anyway? Maybe we should just leave it untouched?

Reply:
i did change the logic more than i supposed to for the PR.
previously, whenever there's a spread + `class:` directive, we always try to set the class regardless whether the directive is dynamic or not.

```svelte
<script>
  let foo = true;
  let obj = { a: 1 };
</script>

<div class:foo {...obj} />
```

in this case none of the attributes are dynamic.

so over here i'm removing the `toggle_class` operation when none of the attributes are dynamic.